src/widget/wwaveformviewer.cpp

```diff
@@ -77,18 +79,29 @@ void WWaveformViewer::mousePressEvent(QMouseEvent* event) {
         m_pScratchPosition->set(targetPosition);
         m_pScratchPositionEnable->slotSet(1.0);
     } else if (event->button() == Qt::RightButton) {
-        // If we are scratching then disable and reset because the two shouldn't
-        // be used at once.
-        if (m_bScratching) {
-            m_pScratchPositionEnable->slotSet(0.0);
-            m_bScratching = false;
+        const auto currentTrack = m_waveformWidget->getTrackInfo();
+        if (!isPlaying() && m_pHoveredMark) {
+            auto cueAtClickPos = getCuePointerFromCueMark(m_pHoveredMark);
+            if (cueAtClickPos) {
+                m_pCueMenuPopup->setTrackAndCue(currentTrack, cueAtClickPos);
+                m_pCueMenuPopup->popup(event->globalPos());
```

Comment:
This is affected by the same bug as #2663, but don't worry about that until we merge #2663.

Reply:
>  We can use the same general approach to add right click menus for beats, downbeats, phrases, and section markers. Hopefully we can reuse some of this code too.

This functionality to find if a point lies in/near a cue marker, beat, downbeat, phrase marker will be quite common. Can this be abstracted to a new parent class?
For this PR, I was thinking of moving this functionality to `WaveformMark` so that it can be commonly used by `WOverview` and `WaveformWidgetRenderer`.